Source/RunActivity/Content/SceneryShader.fx

```diff
@@ -650,6 +1024,29 @@ technique ImageLevel9_3 {
 	}
 }
 
+technique PbrBaseColorMap {
+	pass Pass_0 {
+		//VertexShader = compile vs_4_0_level_9_3 VSGeneral9_3();
+		//PixelShader = compile ps_4_0_level_9_3 PSImage9_3();
+		VertexShader = compile vs_4_0 VSPbrBaseColorMap();
+		PixelShader = compile ps_4_0 PSPbr();
+	}
+}
+
+technique PbrNormalMap {
+	pass Pass_0 {
+		VertexShader = compile vs_4_0 VSNormalMap();
+		PixelShader = compile ps_4_0 PSPbr();
+	}
+}
+
+technique PbrSkinned {
+	pass Pass_0 {
+		VertexShader = compile vs_4_0 VSSkinned();
+		PixelShader = compile ps_4_0 PSPbr();
+	}
+}
```

Comment:
You are using vs_4_0 and ps_4_0 which means that the GPU must support Shader Model 4.
Currently, the minimum required Direct3D feature level of the simulator is 9_3 (some shaders only have 9_3 variants).
With that modification, that means the minimum Direct3D feature level is changing to 10_0. (https://en.wikipedia.org/wiki/Feature_levels_in_Direct3D)

That means we will stop supporting very old GPUs (from 2004 to 2006) which were supporting feature level 9_3 (ATI Radeon Xx00, X1x00 and Nvidia GeForce 6x00, 7x00).
I think it is acceptable, but we have to check with the ORMT if they accept it.

If they accept it, we can upgrade all shaders to SM4 and delete 9_1 variants when they exists (see in the code just below).

Reply:
The limit I faced with the 9_3 was the number of float4 slots possible to pass from the vertex shader to pixel shader, it can be max 8 for 9_3:
https://github.com/pzgulyas/openrails-1/blob/77d23c6e73318948b5d23ed01bdb4efbdf1f66b0/Source/RunActivity/Content/SceneryShader.fx#L258-L269
If going with multiple lights, the lightdir calculation might need to be moved to the pixel shader anyway (which is generally not the best for the performance, because in the vertex shader the calculation needs to be done only 3 times for a triangle, vs. in the pixel shader the pixel number times). Khronos does this in the pixel shader in their gltf reference viewer:
https://github.com/KhronosGroup/glTF-Sample-Viewer/blob/b6d7711fa17733554ce64058185b3480117ac7d7/source/Renderer/shaders/pbr.frag#L159-L171

Reply:
ETCS DMI is 9_3 only.

Reply:
Looking on the history of DriverMachineInterfaceShader.fx, that was originally 9_1, just with the commit f928d2d @cesarBLG modified it to 9_3, but I think that was an error, it doesn't need to be 9_3.

Reply:
I didn't deliberately changed it, probably it was introduced when I rebased the code to official MG version, since it was initially based on NewYear's. I suppose it can be changed back to 9_1.

Reply:
So that means we would no longer support also:
- ATI Radeon 9x00
- nVidia GeForce FX 5x00
- some other GPUs that I expect to have very bad framerates anyway